api/subscriptions/subscriptions.go

```diff
@@ -49,6 +52,11 @@ const (
 )
 
 func New(repo *chain.Repository, allowedOrigins []string, backtraceLimit uint32, txpool *txpool.TxPool) *Subscriptions {
+	cacheSize := backtraceLimit * 120 / 100
+	beat2Cache, _ := newMessageCache(generateBeat2Message, int(cacheSize))
+	beatCache, _ := newMessageCache(generateBeatMessage, int(cacheSize))
+	blockCache, _ := newMessageCache(generateBlockMessage, int(cacheSize))
```

Comment:
Ignoring errors for now, as this will only error if the cache size is less than 0. Should we handle this better?

`subscriptions.New` and `api.New` do not currently return errors

Reply:
I guess we can check if cache is > 0 what is the point of having cache < 0 ?

Reply:
the `backtraceLimit` is already uint32, so it can't be negative

Reply:
I see, maybe just prevent from being 0 then as it makes no sense for 0 cache ?